core/benches/scheduler.rs

```diff
@@ -0,0 +1,494 @@
+
+fn is_tracer<Tx: TransactionWithMeta + Send + Sync + 'static>(tx: &Tx) -> bool {
+    matches!(
+        tx.compute_budget_instruction_details()
+            .requested_loaded_accounts_data_size_limit(),
+        Some(TAG_NUMBER)
+    )
+}
+
+trait TestTxsBuilder {
+    // Tracer is a non-contend low-prio transfer transaction, it'd usually be inserted into the bottom
+    // of Container due to its low prio, but it should be scheduled early for better UX, if its reward
+    // is adequate from bolcker builder perspective.
+    fn build_tracer_transaction(&self) -> RuntimeTransaction<SanitizedTransaction> {
+        let payer = Keypair::new();
+        let to_pubkey = Pubkey::new_unique();
+        let mut ixs = vec![system_instruction::transfer(&payer.pubkey(), &to_pubkey, 1)];
+        ixs.push(ComputeBudgetInstruction::set_compute_unit_price(4));
+        ixs.push(ComputeBudgetInstruction::set_loaded_accounts_data_size_limit(TAG_NUMBER));
+        let message = Message::new(&ixs, Some(&payer.pubkey()));
+        let tx = Transaction::new(&[payer], message, Hash::default());
+        RuntimeTransaction::from_transaction_for_tests(tx)
+    }
+
+    fn build(
+        &self,
+        count: usize,
+        is_single_payer: bool,
+    ) -> Vec<RuntimeTransaction<SanitizedTransaction>> {
+        let mut transactions = Vec::with_capacity(count);
+        // non-contend low-prio tx is first received
+        transactions.push(self.build_tracer_transaction());
+
+        let compute_unit_price = 1_000;
+        let single_payer = Keypair::new();
+        for _n in 1..count {
+            let payer = if is_single_payer {
+                // recreate keypair from single_payer
+                Keypair::from_bytes(&single_payer.to_bytes()).expect("Failed to create Keypair")
+            } else {
+                Keypair::new()
+            };
+            let mut ixs = self.prepare_instructions(&payer);
+            let prioritization =
+                ComputeBudgetInstruction::set_compute_unit_price(compute_unit_price);
+            ixs.push(prioritization);
+            let message = Message::new(&ixs, Some(&payer.pubkey()));
+            let tx = Transaction::new(&[payer], message, Hash::default());
+            let transaction = RuntimeTransaction::from_transaction_for_tests(tx);
+
+            transactions.push(transaction);
+        }
+
+        transactions
+    }
+
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction>;
+}
+
+struct SimpleTransferTxBuilder;
+impl TestTxsBuilder for SimpleTransferTxBuilder {
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction> {
+        let to_pubkey = Pubkey::new_unique();
+        vec![system_instruction::transfer(&payer.pubkey(), &to_pubkey, 1)]
+    }
+}
+
+struct ManyTransfersTxBuilder;
+impl TestTxsBuilder for ManyTransfersTxBuilder {
+    fn prepare_instructions(&self, payer: &Keypair) -> Vec<Instruction> {
+        const MAX_TRANSFERS_PER_TX: usize = 35;
```

Comment:
can we still fit this many with the compute budget instructions that are appended? I think the answer is no (same for the packed noops below).
let's add some sort of check that the actual serialized size is under our expectation - we will want to expand this to use the view types, but if it is larger than a Packet the view types will return an error when we try to construct.